src/main/java/net/caffeinemc/mods/sodium/client/render/SodiumWorldRenderer.java

```diff
@@ -207,20 +206,26 @@ public void setupTerrain(Camera camera,
             this.lastCameraPos = new Vector3d(pos);
         }
 
-        if (this.renderSectionManager.needsUpdate()) {
-            profiler.popPush("chunk_render_lists");
+        boolean runChunkUpdates = true;
 
-            this.renderSectionManager.update(camera, viewport, frame, spectator);
-        }
+        while (runChunkUpdates) {
```

Comment:
Is there anything stopping this from timing out? It should not be able to loop forever.

Reply:
There isn't right now.
I could potentially use `this.renderDistance` as an upper bound on the number of iterations
It shouldn't need more than that in order to visit everything, right?